packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -610,6 +712,69 @@ export abstract class BaseService extends Resource
     this.node.defaultChild = this.resource;
   }
 
+  /**
+   *   Enable Deployment Alarms which take advantage of arbitrary alarms and configure them after service initialization
+  */
+  public enableDeploymentAlarms(alarmConfig: DeploymentAlarmConfig) {
+    // Throw an error if deployment alarms are already configured
+    if (this.deploymentAlarms) {
+      throw new Error('Deployment alarms are already configured.');
+    }
+    // Throw an error if alarms array is empty
+    if (alarmConfig.alarms.length === 0) {
+      throw new Error('Specify at least one deployment alarm');
+    }
+    this.deploymentAlarms = {
+      alarmNames: alarmConfig.alarms.map(alarm => alarm.alarmName),
+      enable: true,
+      rollback: alarmConfig.behavior !== AlarmBehavior.FAIL_ON_ALARM,
+    };
+  }
+
+  /**
+   *   Disassociate existing deployment alarms
+  */
+  public disableDeploymentAlarms() {
+    if (this.deploymentAlarms) {
+      this.deploymentAlarms = {
+        ...this.deploymentAlarms,
+        alarmNames: [],
+        enable: false,
+      };
+    }
+  }
+
+  /**
+   *   Add an alarm based on ECS metric which appended to the list of deployment alarms at synthesis time
+   *   Customize the alarm threshold and evaluationPeriods using 2nd argument ecsMetricAlarmProps.
+   *   `threshold  value 85`
+   *   `evaluationPeriods value 3`
+  */
+  public createEcsMetricAlarm(metric: EcsMetric, ecsMetricAlarmProps?: EcsMetricAlarmProps): cloudwatch.Alarm {
+    // Throw an error if service connect is not configured
+    if (!this._serviceConnectConfig) {
+      throw new Error('Service connect must be enabled to set service connect metric alarms.');
+    }
```

Comment:
I'm not familiar with this part but the code below doesn't seem to be service connect specific. 

the members in `EcsMetric` can be used for regular services though 🤔  right?

Reply:
Sorry I missed this. +1 to Yurui. ABR should not depend on service connect.